app/server/lib/MinIOExternalStorage.ts

```diff
@@ -101,7 +101,7 @@ export class MinIOExternalStorage implements ExternalStorage {
     return this.uploadStream(key, filestream, metadata);
   }
 
-  public async downloadStream(key: string, outStream: stream.Writable, snapshotId?: string ) {
+  public async downloadStream(key: string, snapshotId?: string ): Promise<StreamDownloadResult> {
```

Comment:
What was the motivation for removing `outStream`?

Reply:
It was a less conventional way of using node streams. Most APIs/libraries expect to be receiving a readable stream as a parameter (e.g. the archive libs). 

By using a write output stream, it was introducing unnecessary boilerplate/workarounds that otherwise wouldn't be needed. Using a readable stream with `stream.promises.pipeline` makes for very clean and readable code by comparison. 

I wanted to clean this up now, before these methods saw more widespread use and it would be harder to refactor them.

Reply:
Ok, fair enough. Hope native s3 and azure work same way or close enough.

Reply:
Azure looked like it when I checked!